Packs/GoogleChronicleBackstory/Integrations/GoogleChronicleBackstoryStreamingAPI/GoogleChronicleBackstoryStreamingAPI.py

```diff
@@ -0,0 +1,925 @@
+"""Main file for GoogleChronicleBackstory Integration."""
+from CommonServerPython import *
+
+from typing import Any, Mapping, Tuple, Iterator
+
+from google.oauth2 import service_account
+from google.auth.transport import requests as auth_requests
+from datetime import datetime
+
+''' CONSTANTS '''
+
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%S.%fZ'
+
+SCOPES = ['https://www.googleapis.com/auth/chronicle-backstory']
+MAX_CONSECUTIVE_FAILURES = 7
+
+BACKSTORY_API_V2_URL = 'https://{}backstory.googleapis.com/v2'
+
+ENDPOINTS = {
+    # Stream detections endpoint.
+    'STREAM_DETECTIONS_ENDPOINT': '/detect/rules:streamDetectionAlerts',
+}
+
+TIMEOUT = 300
+MAX_DETECTION_STREAM_BATCH_SIZE = 100
+MAX_DELTA_TIME_FOR_STREAMING_DETECTIONS = '7 days'
+MAX_DELTA_TIME_STRINGS = ['7 day', '168 hour', '1 week']
+IDEAL_SLEEP_TIME_BETWEEN_BATCHES = 30
+IDEAL_BATCH_SIZE = 200
+DEFAULT_FIRST_FETCH = "now"
+
+REGIONS = {
+    "General": "",
+    "Europe": "europe-",
+    "Asia": "asia-southeast1-",
+    "Europe-west2": "europe-west2-"
+}
+
+SEVERITY_MAP = {
+    'unspecified': 0,
+    'informational': 0.5,
+    'low': 1,
+    'medium': 2,
+    'high': 3
+}
+
+MESSAGES = {
+    "NO_RECORDS": 'No Records Found',
+    "INVALID_DATE": 'Invalid {} time, supported formats are: YYYY-MM-ddTHH:mm:ssZ, YYYY-MM-dd, N days, '
+                    'N hours. E.g. 2022-05-15T12:24:36Z, 2021-18-19, 6 days, 20 hours, 01 Mar 2021,'
+                    ' 01 Feb 2021 04:45:33',
+    "INVALID_DELTA_TIME_FOR_STREAMING_DETECTIONS": "First fetch time should not be greater than 7 days or 168 hours (in relative manner compared to current time).",  # noqa: E501
+    "FUTURE_DATE": "First fetch time should not be in the future.",
+    "INVALID_JSON_RESPONSE": 'Invalid response received from Chronicle API. Response not in JSON format.',
+    "INVALID_REGION": 'Invalid response from Chronicle API. Check the provided "Other Region" parameter.',
+    "CONSECUTIVELY_FAILED": 'Exiting retry loop. Consecutive retries have failed {} times.',
+    "PERMISSION_DENIED": 'Permission denied.',
+    "INVALID_ARGUMENTS": "Connection refused due to invalid arguments"
+}
+
+RULE_DETECTION = 'RULE_DETECTION'
+GCTI_FINDING = 'GCTI_FINDING'
```

Comment:
Regarding those, I don't think they should be constants, since you use it only once in specific cases